### PR TITLE
fix: check if tabline is present when calculating certain layouts

### LINKED ISSUE #995: Vertical padding does not take tabline into account
Hi there! Is there any setting to have both ( top and bottom space around telescope ) have equal padding? the bottom padding looks a bit more than the top one.

image

### PR DESCRIPTION
**Pull request by l-kershaw**
- Don't include tabline in the height of the window when present.
  - Only applies to `horizontal`, `vertical` and `center` strategies (also inherited by `flex`)
- ~~Tweaks height calculation for `center` strategy~~

Closes #995

### CODE REVIEW

**Comment by l-kershaw** [comment]
I've removed the tweak for the `center` strategy.
I think I want the tweak to be more precise, and also apply it to some other layouts, so will put it in a separate PR when I get a chance.

**Comment by l-kershaw** [comment]
I've found a problem with the vertical strategy, and some other refactoring to do.
Will hopefully fix today

**Comment by Conni2461** [comment]
Are you still working on this?
My comment would be. "I think i see a reoccurring pattern, maybe a short local `function(max_lines) -> max_lines, tbln`"

Otherwise LGTM (you dont need to do it. I shouldn't be that nitpicky)

File: lua/telescope/pickers/layout_strategies.lua [hunk 1]
```diff
@@ -563,6 +597,12 @@ layout_strategies.vertical = make_documented_layout("vertical", vim.tbl_extend("
     prompt.line = results.line + results.height + 2
   end
 
+  if tbln then
+    prompt.line = prompt.line + 1
+    results.line = results.line + 1
+    preview.line = preview.line ~= nil and preview.line + 1
```
Comment (Conni2461) [inline lua/telescope/pickers/layout_strategies.lua:603]: Hmm why is preview.line only not a thing for vertical. It seems like all other configs set it even if previewer is disabled. (Just asking stupid questions, you know way more about layout)

Comment (l-kershaw) [reply lua/telescope/pickers/layout_strategies.lua]: Not a stupid question, it was bugging me too. Just couldn't figure out a way of refactoring it that didn't seem arbitrary.

Latest commit fixes this (and also allows the `mirror` option to work without a previewer, which will be very useful when we manage to get previewer toggle working).

**Comment by l-kershaw** [comment]
@Conni2461 I have factored out the function you suggested (good spot!), also considered factoring out the "shift down a line if have a tabline", but thought it would be more trouble than it was worth.

I'm happy with this now, but if you could have a last look over before merging, that would be great 😊

**Review by Conni2461 — APPROVED** [review]
LGTM thanks :)

you can ship it if its done. I felt the same about the second refactoring. I also thought it would complicate things.

**Pull request merged**